### PR TITLE
Adding initial docs infra

### PR DESCRIPTION
**Pull request by robscott**
No real content yet, just the initial scaffolding and infra, largely inspired by Gateway API docs. Looks like this right now:

Screenshot 2024-12-19 at 11 54 55 AM

### CODE REVIEW

File: site-src/gieps/giep-116/index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# GIEP-116: GIEP template
```
Comment (ahg-g) [inline site-src/gieps/giep-116/index.md:1]: What is GI? I assume the EP refers to enhancements proposal.

Comment (ahg-g) [reply site-src/gieps/giep-116/index.md:1]: defined below

**Comment by kfswain** [comment]
/lgtm
/approve
/hold

I think it looks good! Most of this is boiler plate. Created: https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/120 for getting the image hosted.

Just holding in case any one else wants to review. @robscott feel free to unhold if you ready to submit this. Thanks a ton!

**Pull request merged**